### PR TITLE
Cosmos DB Output Binding - Authentication tests

### PR DESCRIPTION
**Pull request by berndverst**
# Description

This is part of #958 and adds the initial Cosmos DB Output Binding certification test with complete authentication tests.

The infrastructure changes in this PR have already been applied to the `Dapr Prod` Azure subscription.

### CODE REVIEW

File: tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        name: AzureCosmosDBUrl
+        key:  AzureCosmosDBUrl
+  - name: database
+    secretKeyRef:
+        name: AzureCosmosDB
+        key:  AzureCosmosDB
+  - name: collection
+    secretKeyRef:
+        name: AzureCosmosDBCollection
+        key:  AzureCosmosDBCollection
```
Comment (halspang) [inline tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml:21]: Nit: Do these need to be secrets? I'm fine keeping them that way, but if they weren't it's a bit easier to know where this is pointing. Same question applies to the other cosmosdb yamls.

File: tests/certification/bindings/azure/cosmosdb/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+## Test plan
+
+### Authentication tests
+
+* Authenticate with Azure Active Directory using Service Principal Client Secret
+* Authenticate with Master Key
+
+### Other tests
+
+* Client reconnects (if applicable) upon network interruption
```
Comment (halspang) [inline tests/certification/bindings/azure/cosmosdb/README.md:14]: This is pending yes? Or does it work by nature of the two included tests passing?

File: tests/certification/bindings/azure/cosmosdb/cosmosdb_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/dapr/components-contrib/tests/certification/embedded"
+	"github.com/dapr/components-contrib/tests/certification/flow"
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+)
+
+const (
+	sidecarName = "cosmosdb-sidecar"
+)
+
+func TestKeyVault(t *testing.T) {
```
Comment (halspang) [inline tests/certification/bindings/azure/cosmosdb/cosmosdb_test.go:32]: To clarify, these are really only for making sure an init works, right? Or am I missing something here?

**Review by halspang — COMMENTED** [review]
Just a few questions.

File: tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml (hunk 1 above)
Comment (berndverst) [reply tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml:21]: You just discovered how our conformance test infrastructure works :)

All constants, names and secrets must be provisioned by the `setup-azure-conf-test.sh` script. The values it generates are then exported to KeyVault.

The Conformance Test and Certification Test workflows loads just the required values from KeyVault and injects those into Environment Variables.

To load these Environment Variables I am using the local env var secret store. But to refer to those "environment secrets" I then must specify in the manner seen here.

File: tests/certification/bindings/azure/cosmosdb/README.md (hunk 2 above)
Comment (berndverst) [reply tests/certification/bindings/azure/cosmosdb/README.md:14]: It's TODO. Planning to update this markdown more.

File: tests/certification/bindings/azure/cosmosdb/cosmosdb_test.go (hunk 3 above)
Comment (berndverst) [reply tests/certification/bindings/azure/cosmosdb/cosmosdb_test.go:32]: Right now they are only for making sure the init works with the different authentication methods.

I will be adding more tests, but didn't make this PR too massive.

File: tests/certification/bindings/azure/cosmosdb/README.md (hunk 2 above)
Comment (halspang) [reply tests/certification/bindings/azure/cosmosdb/README.md:14]: I'd prefer if we marked it as TODO here then. I see this file as a quick stop to know what the certification tests are actively/planning on testing.

**Review by halspang — APPROVED** [review]
Small comment which, if you feel strongly opposed to, you don't need to do.

**Review by halspang — APPROVED** [review]

**Review by daixiang0 — APPROVED** [review]

File: tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml (hunk 1 above)
Comment (artursouza) [reply tests/certification/bindings/azure/cosmosdb/components/masterkey/cosmosdb.yaml:21]: I used the same trick for enabling Azure components in E2E tests. This is a nice way to pass values that change per run.

**Review by artursouza — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**